### PR TITLE
Feature/deseng623 - Add frontend, update backend routes to use tenant short name

### PR DESCRIPTION
**Pull request by Baelx**
Issue #: https://github.com/bcgov/met-public/issues/

*Description of changes:*
- Add frontend language view 
- Update backend routes to use tenant short name, add authentication and CORS

*Remaining work:*
- Add tests
- Merge in changes from main
- Create migration to add EDIT_LANGUAGES to admin composite role

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/resources/language.py [hunk 1]
```diff
@@ -101,42 +121,15 @@ def post(language_id, tenant_id):
...
+    @cross_origin(origins=allowedorigins())
+    def delete(language_id, tenant_short_name):
         """Remove a language mapping from a tenant."""
         try:
-            success = LanguageService.remove_language_mapping_from_tenant(language_mapping_id)
-            if success:
-                return 'Successfully deleted language-tenant mapping', HTTPStatus.NO_CONTENT
-            raise ValueError('Language-tenant mapping not found')
+            LanguageService.remove_language_mapping_from_tenant(language_id, tenant_short_name)
+            return {'status': 'success', 'message': 'Tenant language deleted successfully'}, HTTPStatus.OK
```
Comment (NatSquared) [inline met-api/src/met_api/resources/language.py:131]: You could also return [HTTP 204 No Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) to indicate that the resource no longer exists, unless the `message` is planned to be utilized by the app

**Review by NatSquared — APPROVED** [review]
Looking good! I'm excited to see the language system taking shape :) Just a couple optional things to address if you have time

File: met-web/src/components/language/index.tsx [hunk 2]
```diff
@@ -1,7 +1,96 @@
...
+    return (
+        <main style={{ margin: '5%' }}>
+            <Header1>Languages</Header1>
+            <BodyText>Select which languages are available for translations.</BodyText>
+            {languages.length > 0 && (
+                <Autocomplete
+                    style={{ marginTop: '5%', marginBottom: '5%', width: isSmallScreen ? '50%' : '100%' }}
+                    multiple
+                    // Prevent users from clearing all languages with a button click.
+                    clearIcon={<span style={{ display: 'none' }}></span>}
```
[the remark this answers is not in the document]
Comment (Baelx) [reply met-web/src/components/language/index.tsx:83]: Oh, thank you for finding that! I'll be switching

File: met-api/src/met_api/resources/language.py (hunk 1 above)
Comment (Baelx) [reply met-api/src/met_api/resources/language.py:131]: I considered this for sure! Do you think there's a downside to having a message & 200 code returned on delete? I don't necessarily think that a 204 on delete is super idiomatic across the web

**Pull request merged**